packages/remote-client/src/client.ts

```diff
@@ -20,15 +20,17 @@ export class AgentRestClient implements IAgentPlugin {
     url: string
     enabledMethods: string[]
     schema?: IAgentPluginSchema
-    headers?: Record<string, string>
+    headers?: Record<string, string> | (() => Record<string, string>)
```

Comment:
If this function is or can be an async call then it should be signaled in the return type:
```suggestion
    headers?: Record<string, string> | (() => Promise<Record<string, string>>)
```